### PR TITLE
[Hub Generated] Review request for Microsoft.Devices to add version preview/2019-03-22-preview

### PR DESCRIPTION
**Pull request by reshantchandra**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/app/branch/reshantchandra/azure-rest-api-specs/dev-iothub-Microsoft.Devices-2019-03-22-preview...master).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5675 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5675/commits/3a20505bac51cdf6d5e4e208db263fe18478cb46

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/examples/IotHub_CustomerInitiatedFailover.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+              "id": "/subscriptions/ae24ff83-d2ca-4fc8-9717-05dae4bba489/resourceGroups/myResourceGroup/providers/Microsoft.Devices/IotHubs/testHub",
+              "name": "testHub",
+              "type": "Microsoft.Devices/IotHubs",
+              "location": "centraluseuap",
+              "tags": {},
+              "etag": "AAAAAAFD6M4=",
+              "properties": {
+                  "state": "Active",
+                  "provisioningState": "Succeeded",
+                  "ipFilterRules": [],
```
Comment (jlian) [inline specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/examples/IotHub_CustomerInitiatedFailover.json]: Below this we need to add `locations`

```json
    "locations": [
      {
        "location": "West US 2",
        "role": "primary"
      },
      {
        "location": "West Central US",
        "role": "secondary"
      }
    ],
```

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json [hunk 2]
```diff
@@ -0,0 +1,3465 @@
...
+      "description": "The properties of an IoT hub.",
+      "type": "object",
+      "properties": {
+        "authorizationPolicies": {
+          "description": "The shared access policies you can use to secure a connection to the IoT hub.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SharedAccessSignatureAuthorizationRule"
+          }
+        }
```
Comment (jlian) [inline specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: I believe we also need to add details of `locations` here and create a `"$ref": "#/definitions/locations"`

Comment (jlian) [inline specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: Some where here we need to add the definition for `locations`

**Review by jlian — CHANGES_REQUESTED** [review]
I think we might have to add mentions of `locations` in all of the exmples as well

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json [hunk 3]
```diff
@@ -1651,6 +1651,73 @@
...
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false
```
Comment (veronicagg) [inline specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: no need to add this if false.

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json [hunk 4]
```diff
@@ -3407,7 +3407,7 @@
         "outputBlobContainerUri"
       ]
     },
-    "CustomerInitiatedFailoverInput": {
+    "failoverInput": {
```
Comment (veronicagg) [inline specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: Please capitalize definition name.

**Review by veronicagg — CHANGES_REQUESTED** [review]
Added a couple of minor comments inline. 
Have you had the changes reviewed by ARM?

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by jlian** [comment]
> Added a couple of minor comments inline.
> Have you had the changes reviewed by ARM?

@veronicagg given that this is a preview version do we need to have a meeting with ARM or can we just have the review over PR?

**Comment by kapil-ms** [comment]
We need to add changes for message enrichments to this preview version. @reshantchandra

**Comment by veronicagg** [comment]
@reshantchandra @jlian please address comment https://github.com/Azure/azure-rest-api-specs/pull/5544#discussion_r272409931. 
Looks like Ruby SDK and Go SDK are not getting updated, you may update readme.go.md and readme.ruby.md to accommodate the new version. 
Rest looks good. Thanks!

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json (hunk 3 above)
Comment (reshantchandra) [reply specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: removed

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json (hunk 4 above)
Comment (reshantchandra) [reply specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: done

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json (hunk 2 above)
Comment (reshantchandra) [reply specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: Json property for Locations is  NullValueHandling.Ignore hence we don't need it

Comment (reshantchandra) [reply specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: Json property for Locations is  NullValueHandling.Ignore hence we don't need it

**Review by shansin — APPROVED** [review]

**Comment by kapil-ms** [comment]
@montgomp

**Review by veronicagg — APPROVED** [review]
Thanks for addressing the comments!

**Comment by veronicagg** [comment]
@jlian are you ok approving from your side? If not more changes are made and you approve I can merge this PR. Thanks!

**Comment by jlian** [comment]
@reshantchandra have we added the locations property?

**Comment by jlian** [comment]
Oh never mind just saw your comment

**Review by jlian — APPROVED** [review]

**Pull request merged**